### PR TITLE
chore: remove the non-publishing packages from the npm workspace

### PR DESCRIPTION
**Pull request by trentm**
Because they cause difficulties with updating `@opentelemetry/*` dependencies.
This is "Option 3" from #1917

Refs: #1917

# summary of changes

- move all `plugins/node/opentelemetry-instrumentation-FOO/examples/` to `examples/FOO` and fix up some details to get them working.
- move `packages/opentelemetry-sampler-aws-xray` to `incubator/opentelemetry-sampler-aws-xray/`
- These packages are now NOT included in the monorepo workspace.
- Removed `npm run compile:examples`.
- Removed entries for these packages from release-please config and manifest.

# checklist (done)

- [x] Undo "option 1" work from #1928
- [x] Q: is "incubator/" dir a reasonable name? Or put sampler-aws-xray in "archive/"?
- [x] move plugins/node/instr-FOO/examples to top-level examples/FOO dir. (Note that this *reverses* the intent from an effort 1-2y ago to move from examples/ to the instrumentation dirs.)
- [x] Try to confirm that my 'update-otel-deps.js' script now works better at updating deps in this repo.
- [x] manually test each of those examples
- [x] clean up intent in examples/README.md
- [ ] ~start an issue for working through modernizing or dropping examples~
- [ ] ~document expectation/procedure for maintaining examples~

### CODE REVIEW

**Comment by pichlermarc** [comment]
> Q: is "incubator/" dir a reasonable name? Or put sampler-aws-xray in "archive/"?

I think so, yes :) I think it works better than archive and this should be a dir that we can easily re-name (since it's not released, there wouldn't be a lot of links pointing towards this)

> move plugins/node/instr-FOO/examples to top-level examples/FOO dir. (Note that this reverses the intent from an effort 1-2y ago to move from examples/ to the instrumentation dirs.)

That's unfortunate, but I think it's fine. Up-to-date examples are important, but I'd put a higher priority on releasing on time.

**Comment by trentm** [comment]
Sorry for the delay. I'll get to this tomorrow (Wednesday).

**Comment by trentm** [comment]
Here is/was a nice surprise (sarcasm) with npm.

With the above changes the TAV (test-all-versions) tests were failing, but just with `mongodb@4.17.1` (not with 4.17.0 and not with 4.17.2).  What the TAV tests effectively do is: `npm install --no-save mongodb@VERSION; npm test`. Look at what was happening with that `npm install --no-save ...` attempt in the instrumentation-mongodb dir:


```
[~/tm/opentelemetry-js-contrib2/plugins/node/opentelemetry-instrumentation-mongodb (git:tm-1917-option-3)]
% npm ls mongodb
opentelemetry-contrib@0.1.0 /Users/trentm/tm/opentelemetry-js-contrib2
└─┬ @opentelemetry/instrumentation-mongodb@0.39.0 -> ./plugins/node/opentelemetry-instrumentation-mongodb
  └── mongodb@3.6.11

[~/tm/opentelemetry-js-contrib2/plugins/node/opentelemetry-instrumentation-mongodb (git:tm-1917-option-3)]
% npm install --no-save mongodb@4.17.1
...

[~/tm/opentelemetry-js-contrib2/plugins/node/opentelemetry-instrumentation-mongodb (git:tm-1917-option-3)]
% npm ls mongodb
opentelemetry-contrib@0.1.0 /Users/trentm/tm/opentelemetry-js-contrib2
└─┬ @opentelemetry/instrumentation-mongodb@0.39.0 -> ./plugins/node/opentelemetry-instrumentation-mongodb
  └── mongodb@3.6.11
```

`npm install` exited successfully but it did *not* install the request mongodb version in the current dir.

It took a while to notice that while I had removed the "plugins/node/opentelemetry-instrumentation-mongodb/examples" directory, the "package-lock.json" file still had that vestigial entry. Using `npm install --package-lock-only` did *not* remove these entries.

If I manually make this change to that vestigial entry:

```
% git diff -U15
diff --git a/package-lock.json b/package-lock.json
index 89c3fa77..94e12811 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -35060,32 +35060,31 @@
       }
     },
     "plugins/node/opentelemetry-instrumentation-mongodb/examples": {
       "name": "mongodb-example",
       "version": "0.28.0",
       "extraneous": true,
       "license": "Apache-2.0",
       "dependencies": {
         "@opentelemetry/api": "^1.0.0",
         "@opentelemetry/exporter-jaeger": "^1.0.0",
         "@opentelemetry/exporter-zipkin": "^1.0.0",
         "@opentelemetry/instrumentation": "^0.48.0",
         "@opentelemetry/instrumentation-http": "^0.48.0",
         "@opentelemetry/instrumentation-mongodb": "^0.32.0",
         "@opentelemetry/sdk-trace-base": "^1.0.0",
-        "@opentelemetry/sdk-trace-node": "^1.0.0",
-        "mongodb": "^3.6.11"
+        "@opentelemetry/sdk-trace-node": "^1.0.0"
       },
       "devDependencies": {
         "cross-env": "^7.0.3",
         "ts-node": "^10.6.0",
         "typescript": "4.4.4"
       },
       "engines": {
         "node": ">=8.12.0"
       }
     },
     "plugins/node/opentelemetry-instrumentation-mongodb/node_modules/bl": {
       "version": "2.2.1",
       "resolved": "https://registry.npmjs.org/bl/-/bl-2.2.1.tgz",
       "integrity": "sha512-6Pesp1w0DEX1N550i/uGV/TqucVL4AM/pgThFSN/Qq9si1/DF9aIHs1BxD8V/QU0HoeHO6cQRTAuYnLPKq1e4g==",
       "dev": true,
```

Then it works:

```
% npm install --no-save mongodb@4.17.1 && npm ls mongodb
...

% npm ls mongodb
Run `npm audit` for details.
opentelemetry-contrib@0.1.0 /Users/trentm/tm/opentelemetry-js-contrib2
└─┬ @opentelemetry/instrumentation-mongodb@0.39.0 -> ./plugins/node/opentelemetry-instrumentation-mongodb
  └── mongodb@4.17.1 invalid: "3.6.11" from plugins/node/opentelemetry-instrumentation-mongodb

npm ERR! code ELSPROBLEMS
npm ERR! invalid: mongodb@4.17.1 /Users/trentm/tm/opentelemetry-js-contrib2/node_modules/mongodb

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/trentm/.npm/_logs/2024-02-22T19_29_22_304Z-debug-0.log
```


The only things I know to do here are:
1. rm and re-create package-lock.json, or
2. manually edit out those ".../examples" entries from package-lock.json

The latter seems fraught -- there is a possible whole dep tree that I don't
want to try to correctly clean up. So I did the former:

```
% rm package-lock.json
% npm install --package-lock-only
% git diff package-lock.json | wc -l
   51786
```

Sigh.  The result is an inpenetrable 52k line diff.  (IME, npm package-lock updates are unstable on whether some fields are included, e.g.: resolved, integrity, license.  The package-lock stability could be because I'm jumping around on npm versions, but I'm not aware of docs here.)

We *do* see the removal of the vestigial ".../examples/" entries, e.g.:

```diff
-    "plugins/node/opentelemetry-instrumentation-mongodb/examples": {
-      "name": "mongodb-example",
-      "version": "0.28.0",
-      "extraneous": true,
-      "license": "Apache-2.0",
-      "dependencies": {
-        "@opentelemetry/api": "^1.0.0",
-        "@opentelemetry/exporter-jaeger": "^1.0.0",
-        "@opentelemetry/exporter-zipkin": "^1.0.0",
-        "@opentelemetry/instrumentation": "^0.48.0",
-        "@opentelemetry/instrumentation-http": "^0.48.0",
-        "@opentelemetry/instrumentation-mongodb": "^0.32.0",
-        "@opentelemetry/sdk-trace-base": "^1.0.0",
-        "@opentelemetry/sdk-trace-node": "^1.0.0",
-        "mongodb": "^3.6.11"
-      },
-      "devDependencies": {
-        "cross-env": "^7.0.3",
-        "ts-node": "^10.6.0",
-        "typescript": "4.4.4"
-      },
-      "engines": {
-        "node": ">=8.12.0"
-      }
-    },
```

File: .release-please-manifest.json
Comment (trentm) [inline .release-please-manifest.json:1]: reviewer note: This change removes the non-publishing package entries that were added in #1939.

File: examples/express/README.md
Comment (trentm) [inline examples/express/README.md:1]: reviewer note: This diff makes it look like a total re-write, but I only changed the "Installation" section.

File: examples/express/package.json [hunk 1]
```diff
@@ -8,13 +8,12 @@
     "zipkin:client": "cross-env EXPORTER=zipkin ts-node src/client.ts",
     "jaeger:server": "cross-env EXPORTER=jaeger ts-node src/server.ts",
     "jaeger:client": "cross-env EXPORTER=jaeger ts-node src/client.ts",
-    "compile": "tsc -p .",
-    "setup": "cd ../../../../ && npm ci && cd plugins/node/opentelemetry-instrumentation-express && npm run compile && cd examples && npm run compile"
```
Comment (trentm) [inline examples/express/package.json:12]: reviewer note: (/cc @JamieDanielson) this "setup" script is no longer necessary. Now that the examples are not part of the workspace, the user just needs to `npm install`.

File: package-lock.json
Comment (trentm) [inline package-lock.json:1]: reviewer note: Good luck review this. :)  Here is the most succinct version of the diff I could make: https://gist.github.com/trentm/92c639ca7cd86722268d4d4390cecea2

**Review by Flarna — APPROVED** [review]

File: package-lock.json
Comment (trentm) [reply package-lock.json:1]: Just what npm created.

**Comment by trentm** [comment]
> So I'm guessing that test is flaky. I have not repro'd locally yet.

The test failure was on this line:

```
assert.strictEqual(metrics[0].dataPoints[0].value < 1, true);
```

So I wonder if the new `nodejs.event_loop.utilization` metric can unexpectedly be higher than 1. Bug in the math for that metric?

**Review by JamieDanielson — APPROVED** [review]

**Review by pichlermarc — APPROVED** [review]

**Pull request merged**